sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -215,21 +226,12 @@ private boolean shouldSendSessionUpdateForDroppedEvent(
     return false;
   }
 
-  private @Nullable List<Attachment> getAttachments(
-      final @Nullable Scope scope, final @NotNull Hints hints) {
-    List<Attachment> attachments = null;
-    if (scope != null) {
-      attachments = scope.getAttachments();
-    }
-
-    final Object screenshotAttachment = hints.get(SENTRY_SCREENSHOT);
-    if (screenshotAttachment instanceof Attachment) {
-
-      if (attachments == null) {
-        attachments = new ArrayList<>();
-      }
+  private @Nullable List<Attachment> getAttachments(final @NotNull Hints hints) {
+    @NotNull final List<Attachment> attachments = hints.getAttachments().getAll();
 
-      attachments.add((Attachment) screenshotAttachment);
+    @Nullable final Attachment screenshot = hints.getAs(SENTRY_SCREENSHOT, Attachment.class);
```

Comment:
Should screenshot be part of the attachment list instead of a separate key/value in the Map?

Reply:
Code updated